cloudant-client/src/test/java/com/cloudant/tests/HttpTest.java

```diff
@@ -316,27 +324,34 @@ public void cookieRenewal() throws Exception {
                 .password("b")
                 .build();
 
-
         String response = c.executeRequest(Http.GET(c.getBaseUri())).responseAsString();
         assertEquals("The expected response should be received", hello, response);
+
+        // assert that there were 2 calls
+        assertEquals("The server should have received 2 requests", 2, mockWebServer
+                .getRequestCount());
+
+        assertEquals("The request should have been for /_session", "/_session", mockWebServer
+                .takeRequest(10, TimeUnit.SECONDS).getPath());
+        assertEquals("The request should have been for /", "/", mockWebServer
+                .takeRequest(10, TimeUnit.SECONDS).getPath());
+
         String secondResponse = c.executeRequest(Http.GET(c.getBaseUri())).responseAsString();
         assertTrue("There should be no response body on the mock response" + secondResponse,
                 secondResponse.isEmpty());
 
-
         // also assert that there were 3 calls
         assertEquals("The server should have received 3 requests", 3, mockWebServer
                 .getRequestCount());
 
-        mockWebServer.takeRequest(); // cookie
-        mockWebServer.takeRequest(); // actual get
-
         // this is the request that should have the new cookie.
-        RecordedRequest request = mockWebServer.takeRequest();
+        RecordedRequest request = mockWebServer.takeRequest(10, TimeUnit.SECONDS);
```

Comment:
Moved into a static method and added constants in 77b429e (sadly `MockWebServer` is final so I couldn't just override `takeRequest` to always avoid tests hanging for ages).